packages/svelte/src/compiler/phases/3-transform/client/visitors/shared/fragment.js

```diff
@@ -155,8 +154,19 @@ function is_static_element(node) {
 			return false;
 		}
 
-		if (node.name.includes('-')) {
-			return false; // we're setting all attributes on custom elements through properties
+		if (
+			attribute.value !== true &&
+			!is_text_attribute(attribute) &&
+			// if the attribute is not a text attribute but is inlinable we will directly inline it in the
+			// the template so if it's not inlinable we can return false but if it is inlinable (or a text)
+			// attribute we keep count of the inlinable attributes so that if all the attributes are inlinable
+			// we deem the node as static
+			!is_inlinable_expression(
+				Array.isArray(attribute.value) ? attribute.value : [attribute.value],
```

Comment:
I wonder if the method should take `Node | Node[]` so that we can just pass `attribute.value` rather than having to do `Array.isArray(attribute.value) ? attribute.value : [attribute.value]` both places we call it

Reply:
I guess it's a good idea since it will always be called on attributes which could be arrays.

I'll fix it in a bit